### PR TITLE
feat: Added heatmaps opt in check to processing

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We shouldn't process heatmap data where it is explicitly off in the team setting

## Changes

* Adds it to the team manager and checks it before processing

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: plugin-server/src/worker/ingestion/event-pipeline/extractHeatmapDataStep.ts [hunk 1]
```diff
@@ -29,17 +29,21 @@ export function extractHeatmapDataStep(
...
-        const heatmapEvents = extractScrollDepthHeatmapData(event) ?? []
-
-        // eslint-disable-next-line @typescript-eslint/no-floating-promises
-        acks = heatmapEvents.map((rawEvent) => {
-            return runner.hub.kafkaProducer.produce({
-                topic: runner.hub.CLICKHOUSE_HEATMAPS_KAFKA_TOPIC,
-                key: eventUuid,
-                value: Buffer.from(JSON.stringify(rawEvent)),
-                waitForAck: true,
+        const team = await runner.hub.teamManager.fetchTeam(teamId)
```
Comment (pauldambra) [inline plugin-server/src/worker/ingestion/event-pipeline/extractHeatmapDataStep.ts:32]: i'll assume this already does caching

**Review by pauldambra — APPROVED** [review]
about to leave the house, so preemptive tick

**Review by xvello — COMMENTED** [review]
Looks good, missing the unit test for it

File: plugin-server/src/worker/ingestion/event-pipeline/extractHeatmapDataStep.ts (hunk 1 above)
Comment (benjackwhite) [reply plugin-server/src/worker/ingestion/event-pipeline/extractHeatmapDataStep.ts:32]: Yeah it does (probably should be named as such later)

File: plugin-server/tests/worker/ingestion/event-pipeline/extractHeatmapDataStep.test.ts [hunk 2]
```diff
@@ -209,6 +213,12 @@ describe('extractHeatmapDataStep()', () => {
         `)
     })
 
+    it('drops if the associated team has explicit opt out', async () => {
+        runner.hub.teamManager.fetchTeam = jest.fn(() => Promise.resolve({ heatmaps_opt_in: false }))
+        const response = await extractHeatmapDataStep(runner, event)
+        expect(response).toEqual([event, []])
```
Comment (xvello) [inline plugin-server/tests/worker/ingestion/event-pipeline/extractHeatmapDataStep.test.ts:219]: let's add more extensive expects to make sure we don't write to kafka + still strip the event data

```suggestion
        const response = await extractHeatmapDataStep(runner, event)
        expect(response[0]).toEqual(event)
        expect(response[0].properties.$heatmap_data).toBeUndefined()
        expect(response[1]).toHaveLength(0)
        expect(runner.hub.kafkaProducer.produce).toBeCalledTimes(0)
```

**Review by xvello — APPROVED** [review]
test improvement suggestion

**Pull request merged**